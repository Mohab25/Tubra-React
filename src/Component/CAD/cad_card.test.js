import React from 'react'
import '@testing-library/jest-dom'
import { render, fireEvent, screen, getByTestId } from '@testing-library/react'
import CADCard from './cad_card'

describe('testing cad card component ', () => {
    it('renders correctly, __snapshot_test__',()=>{
        const {asFragment} = render(<CADCard/>)
        expect(asFragment()).toMatchSnapshot()
    })
    
    it('calls handle click prop when fireEvent is called',()=>{
        const clickFunc = jest.fn()
        render(<CADCard handleClick={()=>clickFunc(1)}/>)
        fireEvent.click(screen.getByTestId('cad-img'))
        expect(clickFunc).toHaveBeenCalled()
    })
})
