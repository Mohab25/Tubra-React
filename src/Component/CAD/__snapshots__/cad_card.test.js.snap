// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing cad card component  renders correctly, __snapshot_test__ 1`] = `
<DocumentFragment>
  <div
    class="cad-card"
  >
    <div
      class="cad-card-container"
    >
      <div
        class="cad-card-img"
        data-testid="cad-img"
        style="background-image: url(CAD.jpg);"
      />
      <div
        class="cad-card-body"
      >
        <h6>
          <b />
        </h6>
        <p>
          Ipsum ullamco do commodo culpa tempor id qui culpa.
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
